docs/docs/reference-react-component.md

```diff
@@ -6,9 +6,63 @@ category: Reference
+[Components](/react/docs/components-and-props.html) let you split the UI into independent, reusable pieces, and think about each piece in isolation.
+
+## Overview
+
+`React.Component` is an abstract base class, so it rarely makes sense to refer to `React.Component` directly. Instead, you will typically subclass it, and define at least a [`.render()`](#.render) method.
+
+Normally you would define a React component as a plain JavaScript class:
+
+```javascript
+class Greeting extends React.Component {
+  render() {
+    return <h1>Hello, {this.props.name}</h1>;
+  }
+}
+```
+
+If you don't use ES6 yet, you may use the `React.createClass` helper instead:
+
+```javascript
+var Greeting = React.createClass({
+  render: function() {
+    return <h1>Hello, {this.props.name}</h1>;
+  }
+});
+```
+
+The `React.Component` reference assumes the use of ES6. The API for `React.createClass` components is similar with a few exceptions. Take a look at [Using React without ES6](/react/docs/react-without-es6.html) to learn more.
+
+### The Component Lifecycle
+
+Components have three main parts of their lifecycle:
+
+* Mounting: A component is being inserted into the DOM.
+* Updating: A component is being re-rendered to determine if the DOM should be updated.
+* Unmounting: A component is being removed from the DOM.
+
+React provides lifecycle methods that you can specify to hook into this process. Methods prefixed with **`will`** are called right before something happens, and methods prefixed with **`did`** are called right after something happens.
+
+#### Mounting
+
+React will call [`.componentWillMount()`](#.componentwillmount) and [`.componentDidMount()`](#.componentdidmount) when a component is being inserted into the DOM.
+
+#### Updating
+
+State changes can lead to the component being re-rendered. The following methods will be called and are used to determine if the DOM should be updated.  
+
+ - [`.componentWillReceiveProps()`](#.componentwillreceivepropsnextprops) is invoked before a mounted component receives new props and can be used to update state in response to prop changes.
+ - Components will always re-render when the state is updated. Take a look at  [`.shouldComponentUpdate(nextProps, nextState)`](#.shouldcomponentupdatenextprops-nextstate) if you need to change the default behavior.
+ - [`.componentWillUpdate(nextProps, nextState)`](#.componentwillupdatenextprops-nextstate) can be used for preparation before an update occurs.
+ - [`.componentDidUpdate(prevProps, prevState)`](#.componentdidupdateprevprops-prevstate) is invoked after an update and can be used to operate on the DOM after the component has been updated.
+
+#### Unmounting
+
+[`.componentWillUnmount()`](#.componentwillunmount) can be used to perform any cleanup, as it is invoked immediately before a component is unmounted and destroyed.
+
+## Reference
 
   - [`constructor(props)`](#constructor)
```

Comment:
Yeah I agree - let me do that in a subsequent PR though, since this one is merging Hector's content into the layout that I set up.